pimd/pim_mroute.c

```diff
@@ -616,7 +616,7 @@ static int pim_mroute_msg(struct pim_instance *pim, const char *buf,
 
 		connected_src = pim_if_connected_to_source(ifp, ip_hdr->ip_src);
 
-		if (!connected_src) {
+		if (!connected_src && PIM_INADDR_ISNOT_ANY(ip_hdr->ip_src)) {
```

Comment:
The RFC section RFC 3376: 4.2.13 talks only about the IGMP reports. Here the changes are done for all the packets. It is not right, please check.

Reply:
Hi, you're right, but there is already a check in 'igmp_recv_query'
It is about 'non-connected source'